### PR TITLE
util/types: fix truncate float

### PR DESCRIPTION
**Pull request by coocood**
Old implementation truncates 1.36 to 1.35999999999 for column type double(16,2)

### CODE REVIEW

File: util/types/helper.go [hunk 1]
```diff
@@ -37,9 +37,11 @@ func RoundFloat(f float64) float64 {
 // value f to become zero.
 func Round(f float64, dec int) float64 {
 	shift := math.Pow10(dec)
-	f = f * shift
-	f = RoundFloat(f)
-	return f / shift
+	tmp := f * shift
+	if math.IsInf(tmp, 0) {
```
Comment (shenli) [inline util/types/helper.go]: When will we meet this condition?

File: util/types/helper.go [hunk 2]
```diff
@@ -49,37 +51,30 @@ func getMaxFloat(flen int, decimal int) float64 {
...
 	if f > maxF {
 		f = maxF
+		err = ErrOverflow
 	} else if f < -maxF {
 		f = -maxF
+		err = ErrOverflow
 	}
 
-	return f, nil
+	return f, err
```
Comment (shenli) [inline util/types/helper.go]: Trace

File: util/types/helper.go (hunk 1 above)
Comment (coocood) [inline util/types/helper.go]: When f is too large.

**Comment by shenli** [comment]
LGTM
Please make sure that we could pass tidb-test.

**Comment by shenli** [comment]
@tiancaiamao @zimulala PTAL

File: util/types/helper.go (hunk 1 above)
Comment (zimulala) [inline util/types/helper.go]: How about the case that it's true and the decimal value greater than 0?

Comment (coocood) [inline util/types/helper.go]: The float doesn't handle how it is displayed, it should be handled by caller.

**Pull request merged**